viewing_party/party.py

```diff
@@ -1,23 +1,312 @@
+        #                 friends_watched_list.append(user_data["friends"][index1]["watched"][index2])
+        
+        #ENHANCE CODES FOR READABILITY
+        for friend_movie_dicts in (user_data["friends"]): #first loop to give us the movie list (of dictionaries) of "friends"
+
+            for watched_dict in friend_movie_dicts["watched"]: #second loop give us each element (list of dictionaries of "friends' watched list")
+                for key,value in watched_dict.items(): #third loop give us access to each element within the list of friends watched list to compare title
+                    # print(watched_dict["title"])
+                    if watched_dict not in friends_watched_list:
+                        friends_watched_list.append(watched_dict)
         
+        return friends_watched_list
+
+
+def get_user_watched_list(user_data):
+    """Helper function to create user's watch list
+    Input: user data
+    Output: list of user's movie dictionaries
+    
+    """
+    user_watched_list = []
+
+    if len(user_data["friends"]) ==0:
+        user_watched_list = []
+    else:
+        for movie in range(len(user_data["watched"])):
+            user_watched_list.append(user_data["watched"][movie])
+    
+    return user_watched_list
+           
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs(user_data):
+    to_remove_list = get_to_remove_list(user_data)
+
+    user_recs = get_friends_unique_watched(user_data)
+    for element in user_recs[:]:
+        if element in to_remove_list:
+            user_recs.remove(element)
+    return user_recs
+
+
+def get_to_remove_list(user_data):
+    """Helper function to get list of 'subscription' service user do not have
+    Input: user_data
+    Output: list of movie to remove
+    
+    """    
+    friends_unique_list = get_friends_unique_watched(user_data)
+    user_recs = friends_unique_list.copy()
+    to_remove_list = []
+    user_subscription_list = user_data["subscriptions"]
+    for element in range(len(user_recs)):
+        if (user_recs[element]["host"]) not in user_subscription_list:
+            to_remove_list.append(user_recs[element])
+    # print(to_remove_list)
+    return to_remove_list
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+
+def get_new_rec_by_genre(user_data):
+    user_most_popular_genre = get_most_watched_genre(user_data) #get_pop_genre(user_data) 
+    
+    user_recs = get_friends_unique_watched(user_data)
+    recommendations = []
+    
+    for movie in user_recs:
+        if movie["genre"] == user_most_popular_genre:
+            recommendations.append(movie)
+    print(recommendations)
+    return recommendations
+
+def get_rec_from_favorites(user_data):

```

Comment:
The functionality of this function (and some of the other ones) would be a lot cleaner and easier if you start with an empty list and add to it the things you want to return rather than getting a giant list of stuff that you remove things from.

For instance, this function can be simplified to look like this:
```
movies = get_unique_watched(user_data)
    recs = []

    for movie in movies:
        if movie in user_data['favorites']:
            recs.append(movie)

    return recs
```